lib/classes/class-db.php

```diff
@@ -0,0 +1,886 @@
+          $this->_get_file_meta_cache_group($post_id), 
+          false, 
+          $found
+        );
+      }
+
+      /**
+       * Set cache for file meta
+       * 
+       * @param int $post_id
+       * @param string $key
+       * @param mixed $value
+       * @return bool
+       */
+      private function _set_file_meta_cache($post_id, $key, $value) {
+        return wp_cache_set(
+          $this->_get_file_meta_cache_key($post_id, $key), 
+          $value, 
+          $this->_get_file_meta_cache_group($post_id)
+        );
+      }
+
+      /**
+       * Delete file cache
+       * 
+       * @param int $attachment_id
+       */
+      private function _delete_file_cache($attachment_id) {
+        wp_cache_delete( $this->_get_file_cache_key($attachment_id), $this->cache_group );
+      }
+
+      /**
+       * Delete file sizes cache
+       * 
+       * @param int $attachment_id
+       */
+      private function _delete_file_sizes_cache($attachment_id) {
+        wp_cache_delete( $this->_get_file_sizes_cache_key($attachment_id), $this->cache_group );
+      }
+
+      /**
+       * Delete file meta cache for single key
+       * 
+       * @param int $attachment_id
+       * @param string $key
+       */
+      private function _delete_file_meta_key_cache($attachment_id, $key) {
+        wp_cache_delete( 
+          $this->_get_file_meta_cache_key($attachment_id, $key), 
+          $this->_get_file_meta_cache_group($attachment_id) 
+        );
+      }
+
+      /**
+       * Delete all file meta cache
+       * 
+       * @param int $attachment_id
+       */
+      private function _delete_file_meta_cache($attachment_id) {
+        wp_cache_flush_group( $this->_get_file_meta_cache_group($attachment_id) );
+      }
+
+      /**
+       * Delete all attachment cache
+       * 
+       * @param int $attachment_id
+       */
+      private function _delete_attachment_cache($attachment_id) {
+        $this->_delete_file_cache($attachment_id);
+        $this->_delete_file_sizes_cache($attachment_id);
+        $this->_delete_file_meta_cache($attachment_id);
+      }
+
+      /**
+       * Get the total files count known to WP-Stateless
+       * 
+       * @return int
+       */
+      public function get_total_files() {
+        global $wpdb;
+
+        try {
+          $query = "SELECT COUNT(post_id) FROM $this->files";
+
+          return $wpdb->get_var($query);
+        } catch (\Throwable $e) {
+          return 0;
+        }
+      }
+
+      /**
+       * Get file data. If $size_names is set to true, all sizes will be included
+       * 
+       * @param array $meta
+       * @param int $attachment_id
+       * @param bool $with_sizes
+       * @return array
+       */
+      public function get_file($meta, $attachment_id, $with_sizes = false) {
+        if ( defined('WP_STATELESS_POSTMETA') && WP_STATELESS_POSTMETA ) {
+          return get_post_meta($attachment_id, 'sm_cloud', true);
```

Comment:
@balexey88 we have to check if `sm_cloud` exists and is not empty - in this case it should return the value, otherwise we have to try to select data from new tables.

Reply:
I see, we have an additional condition for that. 
It is correct!